it/src/main/java/com/google/cloud/teleport/it/bigquery/DefaultBigQueryResourceManager.java

```diff
@@ -0,0 +1,369 @@
+    write(tableName, ImmutableList.of(row));
+  }
+
+  @Override
+  public synchronized void write(String tableName, List<RowToInsert> rows) {
+    // Exit early if there are no mutations
+    if (!rows.iterator().hasNext()) {
+      return;
+    }
+
+    Table table = getTableIfExists(tableName);
+
+    LOG.info(
+        "Attempting to write {} records to {}.{}.",
+        rows.size(),
+        dataset.getDatasetId().getDataset(),
+        tableName);
+
+    // Send row mutations to the table
+    int successfullyWrittenRecords = rows.size();
+    try {
+      InsertAllResponse insertResponse = table.insert(rows);
+      successfullyWrittenRecords -= insertResponse.getInsertErrors().size();
+
+      if (insertResponse.hasErrors()) {
+        LOG.info("Errors encountered when inserting rows: ");
+        logInsertErrors(insertResponse.getInsertErrors());
+      }
+
+    } catch (Exception e) {
+      throw new BigQueryResourceManagerException("Failed to write to table.", e);
+    }
+
+    LOG.info(
+        "Successfully wrote {} records to {}.{}.",
+        successfullyWrittenRecords,
+        dataset.getDatasetId().getDataset(),
+        tableName);
+  }
+
+  @Override
+  public synchronized ImmutableList<FieldValueList> readTable(String tableName) {
+    Table table = getTableIfExists(tableName);
+
+    // List to store fetched rows
+    ImmutableList.Builder<FieldValueList> immutableListBuilder = ImmutableList.builder();
+
+    LOG.info("Reading all rows from {}.{}", dataset.getDatasetId().getDataset(), tableName);
+
+    // Read all the rows from the table given by tableId
+    try {
+      TableResult results = bigQuery.listTableData(table.getTableId());
+      for (FieldValueList row : results.getValues()) {
+        immutableListBuilder.add(row);
+      }
+    } catch (Exception e) {
+      throw new BigQueryResourceManagerException("Failed to read from table.", e);
+    }
+
+    ImmutableList<FieldValueList> tableRows = immutableListBuilder.build();
+    LOG.info(
+        "Loaded {} rows from {}.{}",
+        tableRows.size(),
+        dataset.getDatasetId().getDataset(),
+        tableName);
+
+    return tableRows;
+  }
+
+  @Override
+  public synchronized void cleanupAll() {
+    LOG.info("Attempting to cleanup manager.");
+    try {
+      if (dataset != null) {
+        Page<Table> tables = bigQuery.listTables(dataset.getDatasetId());
+        for (Table table : tables.iterateAll()) {
+          bigQuery.delete(
+              TableId.of(
+                  projectId, dataset.getDatasetId().getDataset(), table.getTableId().getTable()));
+        }
+        bigQuery.delete(dataset.getDatasetId());
+      }
+    } catch (Exception e) {
+      throw new BigQueryResourceManagerException("Failed to delete resources.", e);
+    }
+    LOG.info("Manager successfully cleaned up.");
+  }
+
+  /** Builder for {@link DefaultBigQueryResourceManager}. */
+  public static final class Builder {
```

Comment:
Similar concerns as for the BigtableResourceManager builder in PR #445:
1. Generate `datasetId` in `build()` only.
2. Remove getters.
3. Remove `setDatasetId`.
4. Don't modify `bigQuery` instance variable state in `build()`.